### DISCUSSION TITLE
How can I add my custom provider correctly? Please Help

### DISCUSSION THREAD

**Discussion by SilvaZhang** [langgenius/dify#11303] [Help] [answered]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I am trying to add my own provider.
Steps:
1. add my provider g2m in api/core/model_runtime/model_provider(copy openai_api_compatible module and do some change)
2. build docker image of api
3. change api image to local in dify/docker/docker-compose.yaml
4. docker compose up -d
5. login to the site and add text embedding model of my provider g2m
6. import document to knowledge

Add provider can be successful, since I can find my provider in settings->model settings, and I can add models successfully.

However, after I import document to knowledge using embedding model, the state of document shows - **Provider g2m does not exist.**

How can I solve this problem? 

### 2. Additional context or comments

Please DO NOT close before I confirm the problem is solved.

**Comment by crazywoola** [comment]
You could try v1.0 later next month, and the process of adding a provider should be easier.

**Comment by SilvaZhang** [comment, marked as answer]
The error is raised from worker when dealing with document text embedding.
I just change the api image to local one in docker-compose.yaml, the image of worker is required as well.
The issue is solved after I change the worker image to the same one with api.
